alpha/lib/model/entry.php

```diff
@@ -1951,11 +1951,53 @@ public function getEntitledKusersEditArray()
 		return array_keys($this->getEntitledUserPuserEditArray());
 	}
 	
+	public function getEntitledKusersView()
+	{
+		return implode(',', array_keys($this->getEntitledPusersViewArray()));
+	}
+
+	public function getEntitledKusersViewArray()
+	{
+		return array_keys($this->getEntitledPusersViewArray());
+	}
+	
 	public function getEntitledPusersEdit()
 	{
 		return implode(',', $this->getEntitledUserPuserEditArray());
 	}
 	
+	public function setEntitledPusersView($v)
+	{
+		$entitledUserPuserView = array();
+		
+		$v = trim($v);
+		if($v == '')
+		{
+			$this->putInCustomData ( "entitledUserPuserView" , serialize($entitledUserPuserView) );
+			return;
+		}
+		
+		$entitledPusersView = explode(',',$v);
+				
+		foreach ($entitledPusersView as $puserId)
+		{
+			$puserId = trim($puserId);
+			if ( $puserId === '' )
+			{
+				continue;
+			}
+
+			$partnerId = kCurrentContext::$partner_id ? kCurrentContext::$partner_id : kCurrentContext::$ks_partner_id;
+			$kuser = kuserPeer::getActiveKuserByPartnerAndUid($partnerId, $puserId);
+			if (!$kuser)
```

Comment:
Do we really want to break the chain if one user is not defined?

Reply:
@MosheMaorKaltura  this behavior is consistent with the way we set co-editors and co-publishers. I don't see a reason this should behave differently.